### DISCUSSION TITLE
Piston explodes although I allow movement everywhere?

### DISCUSSION THREAD

**Discussion by vijaypondini** [GriefPrevention/GriefPrevention#1486] [Q&A]
Hey, I recently observed a player complain of his piston being denied to push, and then exploding ( which is griefprevention's way of stopping it as per what i've seen ), however, in my config, the piston usage is set to everywhere. What can I do to allow this?

PistonMovement: EVERYWHERE
  PistonExplosionSound: true
  

https://user-images.githubusercontent.com/70318494/126652882-1afaead7-ce16-4997-af2d-2e588fffdbc2.mp4

**Comment by Jikoo** [comment]
`EVERYWHERE` is not unrestricted - pistons are not allowed to push into or pull out of claims. Your user is trying to cross a claim boundary with a piston move. If you do not want GP to prevent flying machines entering claims and stealing blocks, you can use `IGNORED`. Blocked pistons are dropped for performance reasons.

See [the wiki](https://github.com/TechFortress/GriefPrevention/wiki/Setup-and-Configuration#pistonmovement) for a writeup of the options. Note that `EVERYWHERE_SIMPLE` does not remove pistons attempting disallowed moves due to how much more performant it is.
